pallets/utxo/src/lib.rs

```diff
@@ -856,38 +856,33 @@ pub mod pallet {
                 inputs.push(TransactionInput::new_empty(*utxo));
             }
 
-            let pubkey_raw: [u8; 32] = match signer.encode().try_into() {
-                Ok(v) => v,
-                Err(e) => {
-                    log::error!("Failed to get caller's public key: {:?}", e);
-                    return Err("Failed to get caller's public key".into());
-                }
-            };
+            let pubkey_raw: [u8; 32] = signer
+                .encode()
+                .try_into()
+                .or_else(|_| Err(DispatchError::Other("Failed to get caller's public key")))?;
 
-            let pubkey: Public = Public::from_raw(pubkey_raw);
-            let wit_prog = match bech32::wit_prog::WitnessProgram::from_address(
-                address[..2].to_vec(),
-                address,
-            ) {
-                Ok(v) => v,
-                Err(e) => {
-                    log::error!("Failed to decode Bech32 address: {:?}", e);
-                    return Err("Invalid Bech32 address".into());
+            let tx_out = match dest {
+                Destination::Pubkey(pubkey) => {
+                    TransactionOutput::new_pubkey(value, H256::from(pubkey))
+                }
+                Destination::ScriptHash(hash) => TransactionOutput::new_script_hash(value, hash),
+                Destination::CreatePP(_, _) | Destination::CallPP(_, _) => {
+                    return Err(
+                        "OP_CREATE/OP_CALL UTXOs cannot be used with `send_to_address`".into(),
+                    )
```

Comment:
Wondering if we could use `dest` directly in the transaction output without this check. Would there be much harm in allowing `CreatePP` and `CallPP` and whatever other spend types we come up in the future to be encoded in a bech32 address?